opensearch/src/test/java/org/opensearch/sql/opensearch/data/type/OpenSearchDataTypeTest.java

```diff
@@ -55,41 +56,96 @@ class OpenSearchDataTypeTest {
 
   private static final OpenSearchDateType dateType = OpenSearchDateType.of(emptyFormatString);
 
+  public static class TestType extends OpenSearchDataType {
+
+    public TestType(MappingType mappingType, ExprCoreType type) {
+      super(mappingType);
+      exprCoreType = type;
+    }
+  }
+
+  @Test
+  public void equals() {
+    assertAll(
+        () -> assertEquals(textType, textType),
+        () -> assertEquals(OpenSearchDataType.of(INTEGER), INTEGER),
+        () -> assertNotEquals(textType, 42),
+        () -> assertEquals(OpenSearchDataType.of(MappingType.GeoPoint),
+            OpenSearchDataType.of(MappingType.GeoPoint)),
+        () -> assertNotEquals(OpenSearchDataType.of(MappingType.GeoPoint),
+            OpenSearchDataType.of(MappingType.Ip)),
+        () -> assertEquals(OpenSearchDataType.of(STRING), OpenSearchDataType.of(STRING)),
```

Comment:
I don't understand the purpose of this test

Reply:
c for coverage
I had to add 4 tests to satisfy jacoco caprise for line 42 and 4 more tests for line 43
https://github.com/Bit-Quill/opensearch-project-sql/blob/5232ad241911135a7e5b852d2b11eb70f0d5fb83/opensearch/src/main/java/org/opensearch/sql/opensearch/data/type/OpenSearchDataType.java#L42-L43